### PR TITLE
CORE-10334 Convert classes in membership module from Kotlin to Java

### PR DESCRIPTION
**Pull request by ac101m**
# Changes
- Re-implemented API interfaces from the membership module in Java.

Corresponding OS PR: https://github.com/corda/corda-runtime-os/pull/3144

### CODE REVIEW

**Review by yarivus — DISMISSED** [review]
LGTM

File: membership/src/main/java/net/corda/v5/membership/EndpointInfo.java
Comment (lankydan) [inline membership/src/main/java/net/corda/v5/membership/EndpointInfo.java]: These docs need fixing (other docs may need it too). Use `{@link EndpointInfo}` I think is the right syntax.

File: membership/src/main/java/net/corda/v5/membership/MemberInfo.java
Comment (lankydan) [inline membership/src/main/java/net/corda/v5/membership/MemberInfo.java:21]: Should use `<pre></pre>` for the code blocks.

**Comment by ac101m** [comment]
@lankydan I've gone over the Jdocs in more detail. Not terribly familiar with the differences between kdocs and javadocs so if I've missed something let me know.

**Comment by ac101m** [comment]
Just waiting for approval from @chrisr3 before merging. PTAL when you get the chance!

File: membership/src/main/java/net/corda/v5/membership/EndpointInfo.java
Comment (chrisr3) [inline membership/src/main/java/net/corda/v5/membership/EndpointInfo.java]: The `:base` and `:application` modules are using
```
<ul>
<li>Java:<pre>{@code ...}</pre></li>
<li>Kotlin:<pre>{@code ...}</pre></li>
</ul>
```
which makes for slightly nicer mark-up. And I think we should be consistent too.

This comment applies to _all_ Javadocs here.

File: membership/build.gradle
Comment (chrisr3) [inline membership/build.gradle:4]: Seeing as you've managed to remove Kotlin _completely_, can you also add:
```
id 'corda.java-only'`
```
here please? This will prevent people adding new Kotlin classes back in.

**Review by chrisr3 — COMMENTED** [review]
Mostly OK, but I think the Javadoc can be tweaked slightly.

**Review by chrisr3 — DISMISSED** [review]
LGTM :+1:

**Comment by ac101m** [comment]
@lankydan Code review from owner required, PTAL.

**Review by lankydan — APPROVED** [review]

**Review by ronanbrowne — APPROVED** [review]
lgtm

**Pull request merged**
